docs/experimental/scope_parser.rst

```diff
@@ -0,0 +1,35 @@
+.. _experimental_scope_parser:
+
+Scope Parser
+============
+
+.. currentmodule:: globus_sdk.experimental.scope_parser
+
+This component defines a Scope object and exposes use of a parser which can
+parse scope strings into trees of Scope objects.
+It should be regarded as very similar to the existing ``MutableScope`` object in
+``globus_sdk.scopes``. Key differences from ``MutableScope``:
+
+- ``Scope.parse`` is available, for parsing scopes from strings
+- The Globus SDK does not support using ``Scope`` in all of the locations where
+  ``MutableScope`` is supported -- Scope objects must be stringified for use
+- ``Scope`` objects define a ``_contains__`` method, allowing one to check if one scope
+  properly contains another
+
+.. warning::
+
+    This module is experimental due to the limitations of scope strings as a
+    representation of consent trees. The scope trees produced by parsing do
+    not necessarily describe real consent structures in Globus Auth.
```

Comment:
> The scope trees produced by parsing do not necessarily describe real consent structures

I don't like this. I think this has been discussed before, but I'm not keen to introduce something that doesn't match the actual Globus Auth implementation.

Reply:
The one thing that I want to have very clearly stated is that it is not possible to represent the consent structures in Auth as a string. Scope string parsing can only match the consent structure produced by a single authentication and consent flow.

The implementation _does_ match what Auth does for a scope string used in a single consent flow. We're just bound by the limitations of the string representation vis-a-vis Auth's more elaborate internal state.